src/test/ui/issues/issue-76547.stderr

```diff
@@ -0,0 +1,35 @@
+error[E0623]: lifetime mismatch
+  --> $DIR/issue-76547.rs:20:13
+   |
+LL | async fn fut(bufs: &mut [&mut [u8]]) {
+   |                          ---------   -
+   |                          |
+   |                          this parameter and the returned future are declared with different lifetimes...
+LL |     ListFut(bufs).await
+   |             ^^^^ ...but data from `bufs` is held across an await point here
+   |
+note: `async fn`s return an `impl Future<Output={return type}`
+  --> $DIR/issue-76547.rs:19:38
+   |
+LL | async fn fut(bufs: &mut [&mut [u8]]) {
+   |                                      ^
```

Comment:
You'll need to handle this case when including the return type snippet to be `impl Future<Output = ()>` for this case.